cmds/sos/sos.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+var (
+	RWLock   sync.RWMutex
+	Registry = make(map[string]uint)
+)
+
+func read(serviceName string) (port uint, exists bool) {
+	RWLock.RLock()
+	defer RWLock.RUnlock()
+	port, exists = Registry[serviceName]
```

Comment:
Is this really returning what you put in the signature?

Reply:
Yes. According to the specs (https://golang.org/ref/spec#Index_expressions), the second resultant is an untyped boolean value, but since exists is a bool, the second result will have type bool

Reply:
I think it's best to return an error type, not a bool; and then if we ever care we can make it support stuff like err.IsExist(). Just a thought.

Reply:
Ok, I will change it in the next PR